Automattic/wp-calypso
# Woop: `/install` software error handling

#### Changes proposed in this Pull Request

### Branched off from https://github.com/Automattic/wp-calypso/pull/59267

This PR implements an error handling for the installing software process of the Woop workflow.

### Most significant changes

* Store the error response when something goes wrong into the state tree
* Check the software installation error in order to update the UI
* Adds a timeout threshold (15 seconds) as a way to  escape unexpected issues


### Testing instructions



### Immediate error

* Testing with an Atomic no Woop site
* Apply this patch D71856-code to generate an error when trying to install the software
* Open the landing page for the Atomic no Woop site:
    `http://calypso.localhost:3000/woocommerce-installation/<your-atomic-not-woop-testing-site>`
* Confirm you get the error page kind of quickly

<img width="600" src="https://user-images.githubusercontent.com/77539/146563869-67a8d8d7-6c20-478f-abe7-c9cd72b0fa0b.png" />

* You can take a look at the POST response:
<img width="600" alt="Screen Shot 2021-12-17 at 11 58 44 AM" src="https://user-images.githubusercontent.com/77539/146563841-78ddc38e-72a9-4efc-9029-c10a1ee746e7.png">

 ### Timeout error
* Testing with an Atomic no Woop site
* Apply this new patch D71860-code
* Open the landing page for the Atomic no Woop site:
    `http://calypso.localhost:3000/woocommerce-installation/<your-atomic-not-woop-testing-site>`
* Confirm you get the error page after 15 seconds
* You can take a look at the network tab to see how the app requests the install software status

https://user-images.githubusercontent.com/77539/146565422-59bff3c2-e138-499d-b446-cbf61a31f70d.mov





Related to https://github.com/Automattic/wp-calypso/issues/58331
